File: backend/src/main/kotlin/bo/edu/umsa/backend/dto/ExecutiveReportDto.kt
package bo.edu.umsa.backend.dto

data class ExecutiveReportDto (
    val totalProjects: Int,
    val completedProjects: Int,
    val completedWithDelayProjects: Int,
    val inProgressProjects: Int,
    val delayedProjects: Int,
    val totalTasks: Int,
    val completedTasks: Int,
    val completedWithDelayTasks: Int,
    val inProgressTasks: Int,
    val pendingTasks: Int,
    val delayedTasks: Int,
    val highPriorityTasks: Int,
    val mediumPriorityTasks: Int,
    val lowPriorityTasks: Int,
    val projectReports: List<ProjectReportDto>,
)

File: frontend/src/app/features/home/models/executive-report.dto.ts
import {ProjectReportDto} from "./project-report.dto";

export interface ExecutiveReportDto {
    completedProjects: number;
    completedTasks: number;
    completedWithDelayProjects: number;
    completedWithDelayTasks: number;
    delayedProjects: number;
    delayedTasks: number;
    highPriorityTasks: number;
    inProgressProjects: number;
    inProgressTasks: number;
    lowPriorityTasks: number;
    mediumPriorityTasks: number;
    pendingTasks: number;
    projectReports: ProjectReportDto[];
    totalProjects: number;
    totalTasks: number;

}
